presto-hive/src/main/java/com/facebook/presto/twitter/hive/thrift/ThriftGeneralInputFormat.java

```diff
@@ -0,0 +1,139 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.twitter.hive.thrift;
+
+import com.hadoop.compression.lzo.LzoIndex;
+import com.twitter.elephantbird.mapred.input.DeprecatedFileInputFormatWrapper;
+import com.twitter.elephantbird.mapreduce.input.MultiInputFormat;
+import com.twitter.elephantbird.mapreduce.io.BinaryWritable;
+import com.twitter.elephantbird.util.TypeRef;
+import io.airlift.units.DataSize;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.LocatedFileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.mapred.FileSplit;
+import org.apache.hadoop.mapred.InputSplit;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.RecordReader;
+import org.apache.hadoop.mapred.Reporter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.facebook.presto.hive.HiveErrorCode.HIVE_INVALID_METADATA;
+import static com.facebook.presto.hive.HiveUtil.checkCondition;
+import static java.lang.Math.toIntExact;
+import static org.apache.hadoop.hive.serde.Constants.SERIALIZATION_CLASS;
+
+/**
+ * Customized version of com.twitter.elephantbird.mapred.input.HiveMultiInputFormat
+ */
+@SuppressWarnings("deprecation")
+public class ThriftGeneralInputFormat extends DeprecatedFileInputFormatWrapper<LongWritable, BinaryWritable>
+{
+    public static final PathFilter lzoSuffixFilter = new PathFilter() {
+        @Override
+        public boolean accept(Path path)
+        {
+            return path.toString().endsWith(".lzo");
+        }
+    };
+
+    public ThriftGeneralInputFormat()
+    {
+        super(new MultiInputFormat());
+    }
+
+    private void initialize(FileSplit split, JobConf job) throws IOException
+    {
+        String thriftClassName = job.get(SERIALIZATION_CLASS);
+        checkCondition(thriftClassName != null, HIVE_INVALID_METADATA, "Table or partition is missing Hive deserializer property: %s", SERIALIZATION_CLASS);
+
+        try {
+            Class thriftClass = job.getClassByName(thriftClassName);
+            setInputFormatInstance(new MultiInputFormat(new TypeRef(thriftClass) {}));
+        }
+        catch (ClassNotFoundException e) {
+            throw new RuntimeException("Failed getting class for " + thriftClassName);
+        }
+    }
+
+    @Override
+    public RecordReader<LongWritable, BinaryWritable> getRecordReader(
+            InputSplit split,
+            JobConf job,
+            Reporter reporter)
+            throws IOException
+    {
+        job.setBoolean("elephantbird.mapred.input.bad.record.check.only.in.close", false);
+        job.setFloat("elephantbird.mapred.input.bad.record.threshold", 0.0f);
```

Comment:
We used to have to configure these often with EB in map reduce. We should make these configurable by table.